doc: Entity management API documentation (#3659)
thin-edge/thin-edge.io: docs/src/operate/registration/index.md

---

> title: Entity Management

Author: Open to use any other title like `Entity Lifecycle Management` that covers CRUD operations on entities. Even the side-pane title (directory name) which is currently `Registration` could be changed accordingly.

tags: [Registration, Deregistration]
sidebar_position: 11
---

import DocCardList from '@theme/DocCardList';

> %%te%% provides two different API interfaces for managing entities on a device:

Reviewer: This introduction is missing something. The focus is on HTTP vs MQTT but the purpose of these APIs is a bit vague. I would give a short overview of these *core functionality*.

```suggestion
%%te%% provides an API for managing entities on a device: to register child devices and services with metadata describing their role and relationship.

%%te%% provides two different flavor of this API:
```

Author: Revamped this intro significantly in 4ff49e7, highlighting the pros and cons of both APIs

---
title: Entity Management
tags: [Registration, Deregistration]
sidebar_position: 11
---

import DocCardList from '@theme/DocCardList';

> %%te%% provides REST APIs to manage all the entities (child devices and services) attached to a main device.
> Entity management includes:
> - Create/Register entities with metadata describing their role and relationship.
> - Fetch the metadata of existing entities
> - Update the metadata of existing entities
> - Delete/Deregister registered entities

Reviewer: A bit too verbose.

```suggestion
%%te%% provides APIs to manage all the entities (child devices and services) attached to a main device.
- Create/Register entities with metadata describing their role and relationship.
- Fetch and update these metadata
- Delete/Deregister entities
```

Author: Resolved by 6963f59

%%te%% provides two different flavors of this API:

- HTTP API: For RESTful communication suitable for request/response patterns.
- MQTT API: For event-driven communication using publish/subscribe patterns.

Both APIs provide access to the same core functionality but are optimized for different use cases.
Choose the most appropriate API based on your specific requirements:
- Use HTTP API for traditional request/response interactions, where the feedback is important
- Use MQTT API for real-time event-driven interactions

For example,the realtime notifications on entity changes make the MQTT APIs ideal for such use-cases,
whereas the inherent feedback in the HTTP APIs make it ideal for operations like querying, deregistration etc,

> where the status of the operation is key.

Author: Resolved by 6963f59
